File: erlang/parhello.erl
-module(parhello).
-author("andrew.pennebaker@gmail.com").
-export([counter/1, hello/1, start/0]).

counter(0) -> init:stop();
counter(N) ->
  receive
    I -> counter(N - I)
  end.

hello(Counter) ->
  receive
    C ->
      io:format("~c", [C]),
      Counter ! 1
  end.

start() ->
  Message = "Hello World!\n",
  Counter = spawn(parhello, counter, [length(Message)]),
  [ spawn(parhello, hello, [Counter]) ! C ||
    C <- Message ].

File: java/ParHello.java
/** Hello World in parallel */
public final class ParHello {
  /** Actor */
  private static class CharPrinter implements Runnable {
    private char c;

    /**
       <p>Construct actor</p>
       @param c character to print
    */
    public CharPrinter(final char c) {
      this.c = c;
    }

    /** Act */
    public void run() {
      System.out.print(c);
    }
  }

  /** utility class */
  private ParHello() {}

  /**
     <p>Main app</p>
     @param args CLI arguments
  */
  public static void main(final String[] args) {
    String message = "Hello World!\n";

    for (char c:message.toCharArray()) {
      new Thread(new CharPrinter(c)).start();
    }
  }
}
